cpp/tests/utilities/column_utilities.cu

```diff
@@ -41,11 +41,12 @@ template <bool check_exact_equality>
 void column_property_comparison(cudf::column_view const& lhs, cudf::column_view const& rhs) {
   EXPECT_EQ(lhs.type(), rhs.type());
   EXPECT_EQ(lhs.size(), rhs.size());
-  EXPECT_EQ(lhs.null_count(), rhs.null_count());
```

Comment:
Why remove the null count check?

Reply:
IMO null count is not a property of a column, it is a property of its content. For example, it is used to compare columns created with `allocate_like`. AFAICT, the column created with `allocate_like` does not have specified content, including the null count. This test was failing with pool mode because the memory is not initialized to zeroes. This looks like a test issue - test was checking something that our API doesn't guarantee.
I went through all places where column_property_comparison is used, and null count was only relevant in JSON test, where values are compared without checking validity.

Reply:
Makes sense to me!